### ISSUE TITLE
[FR] better cooperation with gradle plugin

### ISSUE THREAD

**Issue by alexcohn** [android/ndk#905]
*I'm always frustrated when [...]* I have 'modules' (i.e. subprojects in Android Studio) that have C++ interdependencies. Let me share the most recent example. My app depends on OpenCV, so I set 
````
dependencies {
    implementation 'com.quickbirdstudios:opencv:4.0.1'
}
````
But I also have some C++ code in my app that uses some OpenCV algorithms. All necessary functions are exported from **libopencv_java4.so** which is part of the dependency, but there is no reasonable way I can refer to this prebuilt binary. Gradle will put it deep in `~/.gradle/caches`, not the best location to refer to in my CMakeLists.txt or Android.mk. Eventually, the binary will be copied to `./app/build/intermediates/transforms/mergeJniLibs/debug/0/lib` which is a bit better. But this happens too late: `:externalNativeBuildDebug` is executed earlier.

If I turn off `externalNativeBuild` task, build the project, and then enable the task again – then I can use something like
````
LOCAL_MODULE := opencv
LOCAL_SRC_FILES := $(NDK_OUT)/../../../transforms/mergeJniLibs/debug/0/lib/$(TARGET_ARCH_ABI)/libopencv_java4.so
include $(PREBUILT_SHARED_LIBRARY)
````
Even if I download OpenCV locally and add it as a project dependency, and I know exactly where the native binary is within this module, Gradle does not help me to find its location on disk from CMake or ndk-build scripts. 

I would like to have some way to read the gradle project structure from CMake. E.g. get location of the top project, of the current subproject (the path to build.gradle that launched this cmake), get the list of dependencies…

I can pass some of this info as custom **arguments** for the `externalNativeBuuild`, but I believe that standard convention would help the wider community.

And, again, these workarounds don't resolve the problem I started with: that the native lib becomes *available* too late. I would like to see all native libraries copied to some accessible location as early as possible.

**Comment by alexcohn** [comment]
Another problem related to this scenario. If I have a library module in Android Studio which builds `libnative2.so` and my app builds `libnative1.so` which depends on `libnative2.so`, I cannot use `PREBUILT_SHARED_LIBRARY` (or its CMake equivalent), because on **Sync** step, `libnative2.so` is not there yet.

There should be a documented way to use such binary artefacts from dependent modules, be it downloaded, or built locally.

**Comment by howardpang** [comment]
[https://github.com/howardpang/androidNativeBundle](url) the gradle plugin may help you

**Comment by DanAlbert** [comment]
The primary thing here was fixed by #916. The author of the package needs to expose it properly, but the mechanisms exist (I'm still working on making the authoring process for this smoother). 

The second part (gradle sync interdependency issues) is an AGP problem loosely related to https://issuetracker.google.com/120166563. That's asking specifically about mixing and matching ndk-build and CMake `externalNativeBuild` modules, but the core problem is the same: inter-module native dependencies don't work with gradle sync. I'd also love to see that fixed.
